gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -106,15 +119,60 @@ def self.signature_for_method(method)
     @signatures_by_method[key]
   end
 
+  # when `mod` includes a module with instance methods `method_names`, ensure there is zero intersection between the
+  # final instance methods of `mod` and `method_names`. so, for every m in `method_names`, check if there is already a
+  # method defined on one of `mod`'s `ancestors` with the same name that is final.
+  def self._check_final_ancestors(mod, ancestors, method_names)
```

Comment:
Can we name this such that it's clear that `method_names` are coming from an included module, distinct from the `ancestors` which are the ancestors of the includer? Something like `source` / `target` might again help this distinction.